### ISSUE TITLE
[BUG] Submit-PnpSearchQuery not returning all result rows when specifying -All

### ISSUE THREAD

**Issue by kloosterj** [pnp/powershell#2574]
### Reporting an Issue or Missing Feature
We are trying to use Submit-PnpSearchQuery to find all *.mp4 files on our tenant (150k site collections, 1.9PB), using the following parameters: $Results = Submit-PnpSearchQuery -Query "filetype:mp4" -All

### Expected behavior 
I would expect the returned RowCount would be equal to the TotalRows, because the -All switch was used. Or is there an undocumented limitation? RowCount is not consistent and seems to vary.

### Actual behavior 
When running the query as stated above, RowCount is always less than TotalRows. Recent output:
RowCount                   : 127499
TableType                    : RelevantResults
TotalRows                    : 165323
TotalRowsIncludingDuplicates : 165323

### Steps to reproduce behavior
Connect-PnPOnline -Url https://tenant-admin.sharepoint.com 
$Results = Submit-PnpSearchQuery -Query "filetype:mp4" -All

### What is the version of the Cmdlet module you are running?
1.12.0

### Which operating system/environment are you running PnP PowerShell on?
- [X ] Windows

**Comment by KoenZomers** [comment]
@kloosterj It seems to be working fine for me here. 
<img width="1030" alt="image" src="https://user-images.githubusercontent.com/5940670/204668277-68fbaaa0-23f5-492e-9ebe-32c3a80fbfd6.png">

Can you do some more digging on your end to try to find out when it starts to deviate? Is it with results over 100,000 items for example?

**Comment by kloosterj** [comment]
Whatever I try, it seems to always deviate. Rowcount is always less than TotalRows when searching for FileTypes (where TotalRows will always be large, of course). When searching for a specific FileName though, things are working ok (RowCount = TotalRows = 2, for example).

**Comment by KoenZomers** [comment]
Just tried it again and it always matches here still.

![image](https://user-images.githubusercontent.com/5940670/224453900-0e9c8902-697e-4aca-a3ec-f11f16c3a58b.png)

Perhaps try adding ` -TrimDuplicates:$false` and see if that makes a difference?

**Comment by kloosterj** [comment]
I found the same issue happening with Invoke-MgQuerySearch and raised a ticket with Microsoft about that. Turns out that, for Graph at least, the default sort order is By Relevance. This can cause deviating results when working with Totals on very large result sets. After some fiddling around at aka.ms/SQT we found that adding _"sortlist=[docid]:ascending" and 'refiners='filetype'_ to the graph request, we could achieve pretty consistent total results and a nice refinement result view, that helped us understand how many files of type "mp4" (or any other extension) we have in our environment. 

Guess the same approach can be used with Submit-PnpSearchQuery. See: https://learn.microsoft.com/en-us/sharepoint/dev/general-development/pagination-for-large-result-sets
